SAML certificate issue new with v78.11.0

While performing an upgrade to v78.11.0 from v78.10.0 we started to get this error in the logs:

```
[2026-04-13T18:56:25.949511Z] uaa/ - 10 [main] - [,] .... ERROR --- SpringApplication: Application run failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'identityZoneConfigurationBootstrap' defined in class path resource [org/cloudfoundry/identity/uaa/SpringServletXmlBeansConfiguration.class]: The zone configuration is invalid. You cannot set issuer value unless you have set your own signing key for this identity zone.
```

Our manifest roughly looks like:

```yaml
...
  - name: uaa
    properties:
      bpm:
        enabled: true
      login:
        saml:
          activeKeyId: key-1
          entity_base_url: redacted
          keys:
            key-1:
              certificate: ((uaa_login_saml-key-1.certificate))
              key: ((uaa_login_saml-key-1.private_key))
              passphrase: ""
...
```

Which matches the underlying [spec that hasn't changed in a decade](https://github.com/cloudfoundry/uaa-release/blame/develop/jobs/uaa/spec#L1136-L1255)

On a hunch I created a dev release based on v78.11.0 and removed the changes introduced with https://github.com/cloudfoundry/uaa/pull/3823/changes, deploying that results in a successful deploy.

To support the new `issuer` string, is there a new spec value that needs to be set?

I now understand the issue, so you only have one key.
The SAML legacy structure was misleading at the begin, because the error is not SAML related.

We have for SAML legacy a test in our pipeline but we expect that UAA has only one key defined.

So if I remove this from our test yaml.
https://github.com/cloudfoundry/uaa/blob/develop/scripts/saml/saml-legacy-uaa.yml#L65-L105
, then start UAA, then I see 

Caused by: org.cloudfoundry.identity.uaa.zone.InvalidIdentityZoneConfigurationException: You cannot set issuer value unless you have set your own signing key for this identity zone.


@cweibel You managed it to build a new UAA without a special PR, so you should be able to change your uaa.yaml so that you have a structure like

https://github.com/cloudfoundry/uaa/blob/develop/scripts/saml/saml-legacy-uaa.yml#L65-L105 . In worst case with a copy from the serviceProviderKey section

because I am not sure if should fix it, will try...

--- a/server/src/main/java/org/cloudfoundry/identity/uaa/zone/GeneralIdentityZoneConfigurationValidator.java
+++ b/server/src/main/java/org/cloudfoundry/identity/uaa/zone/GeneralIdentityZoneConfigurationValidator.java
@@ -62,7 +62,7 @@ public class GeneralIdentityZoneConfigurationValidator implements IdentityZoneCo
                     }
                 }
             }
-            if (UaaStringUtils.isNotEmpty(config.getIssuer()) && (tokenPolicy == null || UaaStringUtils.isNullOrEmpty(tokenPolicy.getActiveKeyId()))) {
+            if (!zone.isUaa() && UaaStringUtils.isNotEmpty(config.getIssuer()) && (tokenPolicy == null || UaaStringUtils.isNullOrEmpty(tokenPolicy.getActiveKeyId()))) {
                 throw new InvalidIdentityZoneConfigurationException("You cannot set issuer value unless you have set your own signing key for this identity zone.");
             }
 

--- a/server/src/test/java/org/cloudfoundry/identity/uaa/zone/GeneralIdentityZoneConfigurationValidatorTests.java
+++ b/server/src/test/java/org/cloudfoundry/identity/uaa/zone/GeneralIdentityZoneConfigurationValidatorTests.java
@@ -316,6 +316,43 @@ class GeneralIdentityZoneConfigurationValidatorTests {
                 .hasMessageContaining("You cannot set issuer value unless you have set your own signing key for this identity zone.");
     }
 
+    // Tests for: if (!zone.isUaa() && isNotEmpty(issuer) && (tokenPolicy == null || isNullOrEmpty(activeKeyId)))
+    @MethodSource("parameters")
+    @ParameterizedTest
+    void validate_nonUaaZone_withIssuer_andNoTokenPolicy_throwsException(IdentityZoneValidator.Mode mode) {
+        // non-UAA zone with issuer set but no token policy -> should fail
+        zone.setId("custom-zone");
+        zone.getConfig().setIssuer("http://custom.example.com/issuer");
+        zone.getConfig().setTokenPolicy(null);
+        assertThatThrownBy(() -> validator.validate(zone, mode))
+                .isInstanceOf(InvalidIdentityZoneConfigurationException.class)
+                .hasMessageContaining("You cannot set issuer value unless you have set your own signing key for this identity zone.");
+    }
+
+    @MethodSource("parameters")
+    @ParameterizedTest
+    void validate_nonUaaZone_withoutIssuer_andNoActiveKey_succeeds(IdentityZoneValidator.Mode mode) throws InvalidIdentityZoneConfigurationException {
+        // non-UAA zone without issuer -> condition not triggered, should pass
+        zone.setId("custom-zone");
+        zone.getConfig().setIssuer("http://localhost:8080/uaa");
+        zone.getConfig().setTokenPolicy(null);
+        // validate should not throw
+        assertThatThrownBy(() -> validator.validate(zone, mode))
+            .isInstanceOf(InvalidIdentityZoneConfigurationException.class)
+            .hasMessageContaining("You cannot set issuer value unless you have set your own signing key for this identity zone.");
+    }
+
+    @MethodSource("parameters")
+    @ParameterizedTest
+    void validate_uaaZone_withIssuer_andNoActiveKey_succeeds(IdentityZoneValidator.Mode mode) throws InvalidIdentityZoneConfigurationException {
+        // UAA zone is exempt from the issuer check -> should pass even without an active key
+        zone.setId(IdentityZone.getUaaZoneId());
+        zone.getConfig().setIssuer("http://uaa.example.com/issuer");
+        zone.getConfig().setTokenPolicy(null);
+        // validate should not throw
+        validator.validate(zone, mode);
+    }
+
     @MethodSource("parameters")
     @ParameterizedTest
     void validate_invalid_corsPolicy_xhrConfiguration_allowedUris(IdentityZoneValidator.Mode mode) {

